pylibftdi/driver.py

```diff
@@ -72,36 +73,43 @@ class Driver(object):
     """
 
     # prefer libftdi1 if available. Windows uses 'lib' prefix.
-    _lib_search = {
-        'libftdi': ('ftdi1', 'libftdi1', 'ftdi', 'libftdi'),
-        'libusb': ('usb-1.0', 'libusb-1.0')
+    _lib_search: dict[str, list[str]] = {
+        'libftdi': ['ftdi1', 'libftdi1', 'ftdi', 'libftdi'],
+        'libusb': ['usb-1.0', 'libusb-1.0']
     }
 
-    def __init__(self, libftdi_search=None, **kwargs):
+    def __init__(self, libftdi_search: Optional[str]=None, **kwargs: dict[str, Any]) -> None:
         """
         :param libftdi_search: force a particular version of libftdi to be used
             can specify either library name(s) or path(s)
         :type libftdi_search: string or sequence of strings
         """
         if libftdi_search is not None:
-            self._lib_search['libftdi'] = libftdi_search
+            self._lib_search['libftdi'] = list(libftdi_search)
 
-    def _load_library(self, name, search_list=None):
+        # Library handles.
+        self._fdll: Any = None
+        self._libusb_dll: Any = None
+
+    def _load_library(self, name: str, search_list: Optional[str]=None) -> Any:
         """
         find and load the requested library
 
         :param name: library name
-        :param search_list: sequence or string referring to library names
+        :param search_list: string referring to library names
             library names or paths can be given
         :return: a CDLL object referring to the requested library
         """
+        # If no search list is given, use the default.
+        if not isinstance(search_list, str):
+            raise TypeError(f'library search list must be a string, not {type(search_list)}')
```

Comment:
this is hit if `None` is passed in, confirmed by running `python3 -m pylibftdi.examples.info` - doesn't need a physical FTDI device attached, just libftdi installed. I think this line could / should be removed and just assume search_list is either `None` or `List[str]` (see earlier comments) and either assigned to `libraries` on line 109 (without wrapping in a list) or the search lookup done (line 107)

Reply:
Thanks for catching that. I've added a test in `tests/test_driver.py` that should catch that in the future. The latest commit checks that `search` list is None or a list, otherwise raises a TypeError. Checked that `python3 -m pylibftdi.examples.info` returns the appropriate information.